test/jdk/java/awt/FullScreen/UninitializedDisplayModeChangeTest/DisplayModeChanger.java

```diff
@@ -34,40 +34,45 @@
  */
 public class DisplayModeChanger {
 
-    public static void main(String[] args)
-        throws InterruptedException, InvocationTargetException
-    {
-        final GraphicsDevice gd =
-            GraphicsEnvironment.getLocalGraphicsEnvironment().
-                getDefaultScreenDevice();
+    private static final GraphicsDevice gd =
+            GraphicsEnvironment.getLocalGraphicsEnvironment()
+                               .getDefaultScreenDevice();
+    private static final DisplayMode currentDM = gd.getDisplayMode();
+    private static Frame f = null;
 
-        EventQueue.invokeAndWait(new Runnable() {
-            public void run() {
-                Frame f = null;
+    public static void main(String[] args)
+        throws InterruptedException, InvocationTargetException {
+        try {
+            EventQueue.invokeAndWait(() -> {
                 if (gd.isFullScreenSupported()) {
-                    try {
-                        f = new Frame("DisplayChanger Frame");
-                        gd.setFullScreenWindow(f);
-                        if (gd.isDisplayChangeSupported()) {
-                            DisplayMode dm = findDisplayMode(gd);
-                            if (gd != null) {
-                                gd.setDisplayMode(dm);
-                            }
-                        }
-                        try {
-                            Thread.sleep(1000);
-                        } catch (InterruptedException ex) {
-                            ex.printStackTrace();
-                        }
-                        gd.setFullScreenWindow(null);
-                    } finally {
-                        if (f != null) {
-                            f.dispose();
+                    f = new Frame("DisplayChanger Frame");
+                    gd.setFullScreenWindow(f);
+                    if (gd.isDisplayChangeSupported()) {
+                        DisplayMode dm = findDisplayMode(gd);
+                        if (gd != null) {
+                            gd.setDisplayMode(dm);
                         }
                     }
                 }
+            });
+            try {
+                Thread.sleep(1000);
```

Comment:
Usually I would do Robot.waitForIdle() and then Robot.delay(1000)
But if you say it passed 50 times, then maybe this is enough.